config.go

```diff
@@ -86,13 +89,29 @@ func updateConfNameAndGroup(newdevice string, newgroup string, filePath string)
 	}
 	conf.DeviceName = newdevice
 	conf.Group = newgroup
-	Fs.Remove(filePath)
 	return conf.write()
 }
 
-func writeToHostnameFile(hostname string) error {
-	Fs.Remove(hostnameFile)
-	return afero.WriteFile(Fs, hostnameFile, []byte(hostname), 0644)
+func updateHostnameFiles(hostname string) error {
+	if err := afero.WriteFile(Fs, hostnameFile, []byte(hostname), 0644); err != nil {
+		return err
+	}
+
+	input, err := afero.ReadFile(Fs, hostsFile)
+	if err != nil {
+		return err
+	}
+
+	lines := strings.Split(string(input), "\n")
+
+	for i, line := range lines {
+		if strings.HasPrefix(line, "127.0.0.1") {
```

Comment:
To be more robust you need to check that the line starts with `127.0.0.1` *followed by whitespace*. Without this, lines like `127.0.0.10 foo` would also match here.

You could using `strings.Fields()` to split up the line by whitespace and then check the first element.